android/src/main/java/com/mapbox/mapboxgl/BitmapUtils.java

```diff
@@ -0,0 +1,129 @@
+package com.mapbox.mapboxgl;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.net.Uri;
+import android.util.Base64;
+import android.util.Log;
+import android.util.LruCache;
+import android.view.View;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+
+/**
+ * Created by nickitaliano on 10/9/17.
+ */
+
+public class BitmapUtils {
+  public static final String LOG_TAG = "BitmapUtils";
+
+  private static int CACHE_SIZE = 1024 * 1024;
+  private static LruCache<String, Bitmap> mCache = new LruCache<String, Bitmap>(CACHE_SIZE) {
+    protected int sizeOf(String key, Bitmap bitmap) {
+      return bitmap.getByteCount();
+    }
+  };
+
+  public static Bitmap getBitmapFromURL(String url) {
+    return BitmapUtils.getBitmapFromURL(url, null);
+  }
+
+  public static Bitmap getBitmapFromURL(String url, BitmapFactory.Options options) {
+    Bitmap bitmap = getImage(url);
+
+    if (bitmap != null) {
+      return bitmap;
+    }
+
+    try {
+      InputStream bitmapStream = new URL(url).openStream();
+      bitmap = BitmapFactory.decodeStream(bitmapStream, null, options);
+      bitmapStream.close();
+      addImage(url, bitmap);
+    } catch (Exception e) {
+      Log.w(LOG_TAG, e.getLocalizedMessage());
+      bitmap = Bitmap.createBitmap(1, 1, Bitmap.Config.ALPHA_8); // Returns a transparent bitmap
+    }
+
+    return bitmap;
+  }
+
+  public static Bitmap getBitmapFromResource(Context context, String resourceName, BitmapFactory.Options options) {
+    Resources resources = context.getResources();
+    int resID = resources.getIdentifier(resourceName, "drawable", context.getPackageName());
+    return BitmapFactory.decodeResource(resources, resID, options);
+  }
+
+  public static String createTempFile(Context context, Bitmap bitmap) {
+    File tempFile = null;
+    FileOutputStream outputStream = null;
+
+    try {
+      tempFile = File.createTempFile(LOG_TAG, ".png", context.getCacheDir());
+      outputStream = new FileOutputStream(tempFile);
+    } catch (IOException e) {
+      Log.w(LOG_TAG, e.getLocalizedMessage());
+    }
+
+    if (tempFile == null) {
+      return null;
+    }
+
+    bitmap.compress(Bitmap.CompressFormat.PNG, 100, outputStream);
```

Comment:
Wouldn't using JPG here be more optimal? PNG with a variety of colors can be quite heavy.

Reply:
I made some changes. Do I need to make a new pull request?

Reply:
If it's within the context of this PR then just push to this branch and PR will update

Reply:
It was created from a fork so I can not push it to this branch. Probably I should create a new PR